redis/redlock.go

```diff
@@ -0,0 +1,44 @@
+package redis
+
+import (
+	"context"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-logger/check"
+)
+
+// TODO: update this after LockService interface change (if needed)
+
+const expiry = time.Minute * 30
+
+type redlockWrapper struct {
+	client RedLockClient
+	ctx    context.Context
+}
+
+// NewRedlockWrapper create a new redLock based on a cache instance
+func NewRedlockWrapper(ctx context.Context, client RedLockClient) (*redlockWrapper, error) {
+	if check.IfNil(client) {
+		return nil, ErrNilRedlockClient
+	}
+
+	return &redlockWrapper{
+		client: client,
+		ctx:    ctx,
```

Comment:
it is not go idiomatic to store context variables. Instead, inject them on the functions where are used (if possible like in the `IsBlockProcessed` or `HasConnection` functions)